### DISCUSSION TITLE
Git clone limits (using git commands vs the API)  - what are they?

### DISCUSSION THREAD

**Discussion by chatanooga** [community/community#24841] [Apps, API and Webhooks]
<p>Hi Folks, </p>
<p>If I have a public repo under a free plan what constraints (if any) are there regarding the clone (through git) frequencies? I’m guessing the constraint is not enforced by the API rate-limit one.</p>

**Comment by CleanCut** [comment, marked as answer]
<p>There are currently no specific limits to cloning a repository other than if your usage gets so high that it starts affecting our ability to provide service to others then we will respond in order to keep a good experience for all users.</p>
<p>A (bad) example of a cloning usage pattern that should be avoided is having very large CI jobs doing thousands of clones of different branches of a single repository continuously.</p>
<p>On the other hand, it is perfectly normal for individuals to manually clone a repository whenever they need it, or to set up typical CI jobs that clone and run against a single branch when commits are pushed.</p>

**Reply by bluebrown** [reply]
is cloing the same repo 2 or 3 times per ci job acceptable? We have something like you mentioned with individual branches, but its only 2-3 branches max and using depth 1.

**Reply by CleanCut** [reply]
On a typical small repository, that should be fine.
